specification/devcenter/DevCenter/Environments/models.tsp

```diff
@@ -0,0 +1,226 @@
+
+@doc("Results of the environment list operation.")
+model EnvironmentListResult is Azure.Core.Page<Environment>;
+
+@doc("Properties of an environment.")
+@resource("environments")
+@parentResource(User)
+model Environment {
+  ...EnvironmentUpdateProperties;
+
+  @doc("Environment name.")
+  @key("environmentName")
+  @minLength(3)
+  @maxLength(63)
+  @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_.]{2,62}$")
+  @visibility("read")
+  name: string;
+
+  @doc("Environment type.")
+  @visibility("read", "create")
+  environmentType: string;
+
+  @doc("The AAD object id of the owner of this Environment.")
+  @visibility("read")
+  user?: Azure.Core.uuid;
+
+  @doc("The provisioning state of the environment.")
+  @visibility("read")
+  provisioningState?: EnvironmentProvisioningState;
+
+  @doc("The identifier of the resource group containing the environment's resources.")
+  @visibility("read")
+  resourceGroupId?: string;
+
+  @doc("Name of the catalog.")
+  @visibility("read", "create")
+  catalogName: string;
+
+  @doc("Name of the environment definition.")
+  @visibility("read", "create")
+  environmentDefinitionName: string;
+
+  @doc("Provisioning error details. Populated only for error states.")
+  @visibility("read")
+  error?: Azure.Core.Foundations.Error;
+}
+
+@doc("""
+Properties of an environment. These properties can be updated after the
+resource has been created.
+""")
+model EnvironmentUpdateProperties {
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "there is no build in type to describe object"
+  @doc("Parameters object for the environment.")
+  parameters?: Record<unknown>;
+}
+
+@doc("A catalog.")
+@resource("catalogs")
+@parentResource(Project)
+model Catalog {
+  @doc("Name of the catalog.")
+  @key("catalogName")
+  @minLength(3)
+  @maxLength(63)
+  @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_.]{2,62}$")
+  @visibility("read")
+  name: string;
+}
+
+@doc("Results of the environment definition list operation.")
+model EnvironmentDefinitionListResult is Azure.Core.Page<EnvironmentDefinition>;
+
+@doc("An environment definition.")
+@resource("environmentDefinitions")
+@parentResource(Catalog)
+model EnvironmentDefinition {
+  @doc("The ID of the environment definition.")
+  id: string;
+
+  @doc("Name of the environment definition.")
+  @key("definitionName")
+  @minLength(3)
+  @maxLength(63)
+  @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_.]{2,62}$")
+  @visibility("read")
+  name: string;
+
+  @doc("Name of the catalog.")
+  catalogName: string;
+
+  @doc("A short description of the environment definition.")
+  description?: string;
+
+  @doc("Input parameters passed to an environment.")
+  parameters?: EnvironmentDefinitionParameter[];
+
+  #suppress "@azure-tools/typespec-azure-core/no-unknown" "there is no build in type to describe json"
+  //using unkown here is better than string since it generates BinaryData in the SDK
+  @doc("JSON schema defining the parameters object passed to an environment.")
+  parametersSchema?: unknown;
```

Comment:
does this property also accept primitve values? Like string, int, etc? If not, it might be better to make this one Record<unknown>

Reply:
no, there is no primitive type. I've updated to `Record<unknown>`, but it's actually incorrect. Let's say  `parametersSchema :  "{\"type\":\"object\"}"`. What is inside the bracket is map<string, T>, but the whole input inclusing the brackets is actually an object. Both model validation and sdk serialization fails when using that. I've reverted to `unknown`

Reply:
We shouldnt be defining the TypeSpec to optimize for SDK generation, instead we should have the TypeSpec represent the correct type for the service.

@m-nash @chunyu3 @johanste thoughts on the correct type here?

Reply:
If it is a (json) object (e.g. `"type": "object"` in openapi/swagger, then the correct type is `Record<unknown>`.

Reply:
If this literally could be any keyed json object, then `Record<unknown>` is the correct representation.

Reply:
Per the discussion on teams, seems we dont have a way to accurately represent this today. Tim opened an issue to consider adding support for this type of field in the future: https://github.com/microsoft/typespec/issues/3005

`string` would be more accurate with what the service currently defines for this property. We dont have a way to constrain unknown here so we might not be representing this property correctly if we keep that.